### PR TITLE
Update msal to v2

### PR DESCRIPTION
**Pull request by millerds**
Thank you for your pull request!  Please provide the following information.

---

**Change Description**:
Updating msal version to new library package and to v2.  Required object changes were made as a result.
Also updated some VS Code launch.json settings related to sso specific npm commands and
tweaked the taskpane code to more closely match the other templates.

1. **Do these changes impact any *npm scripts* commands (in package.json)?** (e.g., running 'npm run start')
No

2. **Do these changes impact *VS Code debugging* options (launch.json)?**
Yes.  Same options, just had to adjust what npm script was called.

3. **Do these changes impact *template output*?** (e.g., add/remove file, update file location, update file contents)
Yes.  The code output had different objects in the fallback auth dialog relating to MSAL.

4. **Do these changes impact *documentation*?** (e.g., a tutorial on https://docs.microsoft.com/en-us/office/dev/add-ins/overview/office-add-ins)
Possibly . . . if there is any explanation of MSAL calling they will be out of date.

If you answered yes to any of these please do the following:
    > Include 'Rick-Kirkham' in the review
    > Make sure the README file is correct

**Validation/testing performed**:
Ran template using dev tenant and Microsoft account (verifying MFA)

### CODE REVIEW

File: src/helpers/fallbackauthdialog.ts [hunk 1]
```diff
@@ -44,23 +49,18 @@ const msalConfig: Msal.Configuration = {
...
-const userAgentApp: Msal.UserAgentApplication = new Msal.UserAgentApplication(msalConfig);
+const publicClientApp: PublicClientApplication = new PublicClientApplication(msalConfig);
 
-function authCallback(error, response) {
-  if (error) {
-    console.log(error);
-    Office.context.ui.messageParent(JSON.stringify({ status: "failure", result: error }));
+function handleResponse(response) {
+  if (response.tokenType === "id_token") {
+    console.log(response.idToken.rawIdToken);
```
Comment (davidchesnut) [inline src/helpers/fallbackauthdialog.ts:60]: Don't log out the token as it contains PII.

File: src/helpers/fallbackauthdialog.ts [hunk 2]
```diff
@@ -44,23 +49,18 @@ const msalConfig: Msal.Configuration = {
   },
 };
 
-var requestObj: Object = {
+var requestObj: RedirectRequest = {
```
Comment (davidchesnut) [inline src/helpers/fallbackauthdialog.ts:52]: You need to request Graph scopes as "extraScopesToConsent". You don't want the access token to actually have Graph scopes from the client side. The server will request that later through the OBO flow.

**Review by davidchesnut — CHANGES_REQUESTED** [review]
Looks good. I noted a couple things that were feedback items Dan Saunders gave me at one point.

File: src/helpers/fallbackauthdialog.ts (hunk 1 above)
Comment (millerds) [reply src/helpers/fallbackauthdialog.ts:60]: BTW . . . the sso sample you've been working on also logs this out :-).

File: src/helpers/fallbackauthdialog.ts (hunk 2 above)
Comment (millerds) [reply src/helpers/fallbackauthdialog.ts:52]: Can you explain a little more?  I was just changing the msal library being used and keeping the rest of it as it already was.  This sounds like something that needs to be done as part of the other change I'm making to move the token handling to middle-tier . . . but I'm not sure what changes to the scope should look like or why.

Comment (davidchesnut) [reply src/helpers/fallbackauthdialog.ts:52]: Here's an example of [how to use it](https://github.com/OfficeDev/Office-Add-in-samples/blob/5668fa059457f218570f4347f67d859383b5736e/Samples/auth/Office-Add-in-NodeJS-SSO/Complete/public/javascripts/fallback-msal/authConfig.js#L14) Would be nice if you could pick up the change here, but if you are covering it in a separate PR that works. Dan recommended this pattern to me so he knows more about the implementation details and the risk.

File: src/helpers/fallbackauthdialog.ts (hunk 1 above)
Comment (davidchesnut) [reply src/helpers/fallbackauthdialog.ts:60]: It shouldn't, if you can point out where that is happening, would be super useful and I will fix it! Thanks!

Comment (millerds) [reply src/helpers/fallbackauthdialog.ts:60]: authRedirect.js L24

File: src/helpers/fallbackauthdialog.ts (hunk 2 above)
Comment (millerds) [reply src/helpers/fallbackauthdialog.ts:52]: I can do it in either spot (the template won't be enabled until they are both in).

I've been looking at your sample as an example of where I should be making changes and I saw the scope that you had . . . however I'm not sure what I should put for extraScopesToConsent for my case (I'm not reading files, I'm reading user profile information) and I'm not sure if I should be changing the scopes property here as well.

**Review by Rick-Kirkham — APPROVED** [review]

**Review by igor-ribeiiro — APPROVED** [review]

File: src/helpers/fallbackauthdialog.ts (hunk 1 above)
Comment (millerds) [reply src/helpers/fallbackauthdialog.ts:60]: Actually . . . I was confusing two calls . . . please disreguard.

File: .vscode/tasks.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
     {
       "label": "Debug: Excel Desktop",
       "type": "npm",
-      "script": "start:desktop -- --app excel",
```
[the remark this answers is not in the document]
Comment (millerds) [reply .vscode/tasks.json:34]: SSO doesn't have an option to do "start:desktop" with the reason being it's not using the same start debugging call all the other projects are using, it's using an sso one that isn't as full featured.  We need to invest in reconciling the two start methods before adding these back (they didn't work as is).

File: src/helpers/fallbackauthdialog.html [hunk 4]
```diff
@@ -10,7 +10,7 @@
 
     <!-- Office JavaScript API -->
     <script type="text/javascript" src="https://ajax.microsoft.com/ajax/4.0/MicrosoftAjax.js"></script>
-    <script type="text/javascript" src="https://alcdn.msauth.net/lib/1.1.3/js/msal.min.js"></script>
+    <script type="text/javascript" src="https://alcdn.msauth.net//browser/2.26.0/js/msal-browser.min.js"></script>
```
[the remark this answers is not in the document]
Comment (millerds) [reply src/helpers/fallbackauthdialog.html:13]: Yup :-)

File: src/helpers/fallbackauthdialog.ts (hunk 2 above)
Comment (millerds) [reply src/helpers/fallbackauthdialog.ts:53]: I'll look into this in the other change I working on.

**Pull request merged**